### ISSUE TITLE
Multiline ComboBox

### ISSUE THREAD

**Issue by koffeinfrei** [carbon-design-system/carbon-components-svelte#1176]
There's currently no component that support multiline content in a combobox / dropdown. I couldn't find anything pointing in this direction on https://www.carbondesignsystem.com either.

A potential solution could be to introduce a variant for the `ComboBox` component. I played around with it, and the following would roughly cover my current use case (being a 3-line item entry):

```diff
diff --git a/node_modules/carbon-components-svelte/src/ComboBox/ComboBox.svelte b/node_modules/carbon-components-svelte/src/ComboBox/ComboBox.svelte
index be693f3..f92fcae 100644
--- a/node_modules/carbon-components-svelte/src/ComboBox/ComboBox.svelte
+++ b/node_modules/carbon-components-svelte/src/ComboBox/ComboBox.svelte
@@ -363,7 +363,7 @@
               highlightedIndex = i;
             }}"
           >
-            {itemToString(item)}
+            {@html itemToString(item)}
             {#if selectedItem && selectedItem.id === item.id}
               <Checkmark16 class="bx--list-box__menu-item__selected-icon" />
             {/if}
```

It also requires some additional css, but that doesn't necessarily have to live in [carbon-components-svelte](https://github.com/carbon-design-system/carbon-components-svelte).

```scss
.bx--list-box__menu-item,
.bx--list-box__menu-item__option {
  // carbon-components/scss/globals/scss/vendor/@carbon/styles/scss/components/list-box/_list-box.scss
  height: rem(2 * 40px);
  line-height: 1.2rem;
}
```

The result is such we can have something like this:
![image](https://user-images.githubusercontent.com/176582/158346860-3e19f052-d8c1-496d-a08b-ad401d3cbc6b.png)

The minimal change in [carbon-components-svelte](https://github.com/carbon-design-system/carbon-components-svelte) to support my use case would be to allow `@html` as the item content.

What do you think about something like this? Do you have any other plans? Is there another way to achieve this?

**Comment by koffeinfrei** [comment]
> I'm inclined to use slots over @html so the consumer has full control.

That's even better :pray:

**Comment by metonym** [comment]
Released in [v0.62.0](https://github.com/carbon-design-system/carbon-components-svelte/releases/tag/v0.62.0).

**Single-line, custom slot**

```svelte
<ComboBox
  titleText="Contact"
  placeholder="Select contact method"
  items={[
    { id: "0", text: "Slack" },
    { id: "1", text: "Email" },
    { id: "2", text: "Fax" },
  ]}
  let:item
  let:index
>
  {item.text} {item.id} {index}
</ComboBox>
```

[**Multi-line, custom slot**](https://carbon-components-svelte.onrender.com/components/ComboBox#custom-slot)

To achieve a multi-line row, you will need to override the `height` of `.bx--list-box__menu-item, .bx--list-box__menu-item__option` (it seems you are already doing this based on the screenshot provided).

```svelte
<ComboBox
  titleText="Contact"
  placeholder="Select contact method"
  items={[
    { id: "0", text: "Slack" },
    { id: "1", text: "Email" },
    { id: "2", text: "Fax" },
  ]}
  let:item
  let:index
>
  <div>
    <strong>{item.text}</strong>
  </div>
  <div>
    id: {item.id} - index:
    {index}
  </div>
</ComboBox>

<style>
  :global(.bx--list-box__menu-item, .bx--list-box__menu-item__option) {
    height: auto;
  }
</style>
```
